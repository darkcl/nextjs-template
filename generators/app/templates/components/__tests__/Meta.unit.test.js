import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Meta from '../Meta';

describe('<Meta />', () => {
  it('should render correctly with no properties', () => {
    const meta = shallow(<Meta />);
    expect(toJson(meta)).toMatchSnapshot();
  });

  it('should render correctly with only title', () => {
    const meta = shallow(<Meta title="Page Title"/>);
    expect(toJson(meta)).toMatchSnapshot();
  });

  it('should render correctly with only description', () => {
    const meta = shallow(<Meta description="Page Description"/>);
    expect(toJson(meta)).toMatchSnapshot();
  });
});
