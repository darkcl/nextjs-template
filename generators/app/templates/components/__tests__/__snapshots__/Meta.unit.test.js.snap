// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta /> should render correctly with no properties 1`] = `<SideEffect(Head) />`;

exports[`<Meta /> should render correctly with only description 1`] = `
<SideEffect(Head)>
  <meta
    content="Page Description"
    name="description"
  />
</SideEffect(Head)>
`;

exports[`<Meta /> should render correctly with only title 1`] = `
<SideEffect(Head)>
  <title>
    Page Title
  </title>
</SideEffect(Head)>
`;
